File: spring-boot-VMM/src/main/java/pp/project/vmm/config/security/rest/response/JwtResponse.java
package pp.project.vmm.config.security.rest.response;

import lombok.*;

import java.util.List;
import java.util.UUID;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class JwtResponse {


    private UUID id;
    private String username;
    private List<String> roles;
    private String accessToken;
    private String tokenType;

    public JwtResponse(String jwt, UUID id, String username, List<String> roles) {
        this.accessToken = jwt;
        this.id = id;
        this.username = username;
        this.roles = roles;
        this.tokenType = "Bearer";
    }

}

File: angular-VMM/src/app/auth/JwtResponse.ts
export class JwtResponse{
    id:string;
    username:string;
    roles: string[];
    accessToken: string;
    tokenType: string;


    constructor(id: string, username: string, roles: string[], accessToken: string, tokenType: string) {
        this.id = id;
        this.username = username;
        this.roles = roles;
        this.accessToken = accessToken;
        this.tokenType = tokenType;
    }
}
